spec/plans/guest-topology.fmf

```diff
@@ -0,0 +1,114 @@
+summary: Define format of on-disk description of provisioned guest topology
+title: Guest topology format
+order: 95
+
+description: |
+    The following text defines structure of files tmt uses for exposing
+    guest names, roles and other properties to tests and steps that run
+    on a guest (:ref:`prepare </spec/plans/prepare>`,
+    :ref:`execute </spec/plans/execute>`, :ref:`finish </spec/plans/finish>`).
+    tmt saves these files on every guest used, and exposes their paths to
+    processes started by tmt on these guests through environment variables:
+
+    * ``TMT_TOPOLOGY_YAML`` for a YAML file
+    * ``TMT_TOPOLOGY_BASH`` for a shell-friendly ``NAME=VALUE`` file
+
+    Both files are always available, and both carry the same information.
+
+    .. warning::
+
+        The shell-friendly file contains arrays, therefore it's compatible
+        with Bash 4.x and newer.
+
+    .. note::
+
+        The shell-friendly file is easy to ingest for a shell-based tests,
+        it can be simply ``source``-ed. For parsing the YAML file in shell,
+        pure shell parsers like https://github.com/sopos/yash can be used.
+
+    .. code-block:: yaml
+        :caption: TMT_TOPOLOGY_YAML
+
+        # Guest on which the test or script is running.
+        guest:
+            name: ...
+            role: ...
+            hostname: ...
+
+        # List of names of all provisioned guests.
+        guest_names:
```

Comment:
So far we're consistently using `-` for separating multiword keys in `yaml` files (except a few exceptions which slipped through :P ) Could we continue in that direction here as well?

Reply:
Fixed in 2526d401.

Reply:
Thanks. The spec example still uses `guest_names` and `role_name` though.

Reply:
Fixed in 9950ce74.